### ISSUE TITLE
3-node multi-master quits serving over 8080 and logs

### ISSUE THREAD

**Issue by yebo29** [victorcoder/dkron#426]
I have a 3 node multi-master setup and every once in a while dkron stops serving the UI over port 8080, requiring a restart of the dkron service to bring the UI back up. I'm unable to determine if jobs continue to execute on the affected nodes. 
Is this by some design or misconfiguration or bug? See below configs:

Bootstrap node:
`{
  "backend": "etcd",
  "backend_machine": "xxx.xxxx.xxx:2379",
  "server": true,
  "log-level": "debug",
  "tags": {
    "role": "master",
    "datacenter": "east"
  },
  "keyspace": "dkron",
  "encrypt": "xxxxxxx",
  "join": [],
  "mail_host": "email-smtp.us-east-1.amazonaws.com",
  "mail_port": 587,
  "mail_username": "xxxxx",
  "mail_password": "xxxxx",
  "mail_from": "cron@xxxx.com",
  "mail_subject_prefix": "[Dkron]"
}`

The other two nodes:
`{
  "backend": "etcd",
  "backend_machine": "xxx.xxxx.xxx:2379",
  "server": true,
  "log-level": "debug",
  "tags": {
    "role": "master",
    "datacenter": "east"
  },
  "keyspace": "dkron",
  "encrypt": "xxxxxxx",
  "join": ["dkron.xxxx.com"],
  "mail_host": "email-smtp.us-east-1.amazonaws.com",
  "mail_port": 587,
  "mail_username": "xxxxx",
  "mail_password": "xxxxx",
  "mail_from": "cron@xxxx.com",
  "mail_subject_prefix": "[Dkron]"
}`

All that I see in syslog is:

grep -i dkron /var/log/syslog
Sep  6 12:52:01 ip-10-0-30-51 systemd[1]: Stopping Dkron Agent...
Sep  6 12:52:01 ip-10-0-30-51 systemd[1]: Stopped Dkron Agent.
Sep  6 12:52:01 ip-10-0-30-51 systemd[1]: Starting Dkron Agent...
Sep  6 12:52:01 ip-10-0-30-51 systemd[1]: Started Dkron Agent.

Also, I set the log level to 'debug' but still I don't see much in the syslog. Are they stored elsewhere? Is there a way to specify a log file location? Thanks!

**Comment by yebo29** [comment]
Running version 0.10.4

**Comment by yebo29** [comment]
Disregard logs question, just saw https://github.com/victorcoder/dkron/issues/395

**Comment by yebo29** [comment]
Update: dkron processes keep adding up as well i.e. they do not terminate after a stop or restart command from systemd. I have to manually kill -9 them. 

```
dkron     4617  0.2  3.5 341396 35872 ?        Sl   Oct04   4:04 /usr/bin/dkron agent -server
dkron     4622  0.0  1.1  24288 11456 ?        Sl   Oct04   0:01 /usr/bin/dkron-processor-files
dkron     4631  0.0  1.1  24288 11728 ?        Sl   Oct04   0:01 /usr/bin/dkron-processor-log
dkron     4639  0.0  1.1  24308 12036 ?        Sl   Oct04   0:01 /usr/bin/dkron-processor-syslog
dkron     4647  0.0  1.2  26772 12376 ?        Sl   Oct04   0:01 /usr/bin/dkron-executor-shell
dkron    15347  0.0  1.1  22044 11316 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-files
dkron    15356  0.0  1.1  22044 11432 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-log
dkron    15364  0.0  1.1  22064 11444 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-syslog
dkron    15587  0.0  1.1  23100 11944 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-files
dkron    15596  0.0  1.2  23100 12500 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-log
dkron    15604  0.0  1.2  23120 12528 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-syslog
dkron    15612  0.0  1.1  26772 11856 ?        Sl   Oct02   0:02 /usr/bin/dkron-executor-shell
dkron    21596  0.0  1.1  22044 11376 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-files
dkron    21605  0.0  1.1  22044 11420 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-log
dkron    21613  0.0  1.1  22064 11376 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-syslog
dkron    21621  0.0  1.1  25716 11792 ?        Sl   Oct02   0:02 /usr/bin/dkron-executor-shell
dkron    23192  0.0  1.1  22044 11316 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-files
dkron    23201  0.0  1.1  22044 11420 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-log
dkron    23209  0.0  1.1  22064 11444 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-syslog
dkron    23217  0.0  1.1  26772 11720 ?        Sl   Oct02   0:02 /usr/bin/dkron-executor-shell
dkron    24040  0.0  1.1  22044 11384 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-files
dkron    24049  0.0  1.1  22044 11424 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-log
dkron    24057  0.0  1.1  22064 11444 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-syslog
dkron    24065  0.0  1.1  24308 11520 ?        Sl   Oct02   0:02 /usr/bin/dkron-executor-shell
dkron    25591  0.0  1.1  22044 11320 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-files
dkron    25600  0.0  1.1  22044 11484 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-log
dkron    25608  0.0  1.1  22064 11444 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-syslog
dkron    25616  0.0  1.1  22064 11400 ?        Sl   Oct02   0:00 /usr/bin/dkron-executor-shell
dkron    25738  0.0  1.1  22044 11312 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-files
dkron    25747  0.0  1.1  22044 11424 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-log
dkron    25755  0.0  1.1  22064 11440 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-syslog
dkron    25763  0.0  1.1  26772 11852 ?        Sl   Oct02   0:02 /usr/bin/dkron-executor-shell
dkron    27146  0.0  1.1  22044 11384 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-files
dkron    27155  0.0  1.1  22044 11420 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-log
dkron    27163  0.0  1.1  22064 11372 ?        Sl   Oct02   0:00 /usr/bin/dkron-processor-syslog
dkron    28381  0.0  1.1  24288 12112 ?        Sl   Oct02   0:02 /usr/bin/dkron-processor-files
dkron    28391  0.0  1.1  24288 11820 ?        Sl   Oct02   0:02 /usr/bin/dkron-processor-log
dkron    28399  0.0  1.1  24308 12108 ?        Sl   Oct02   0:02 /usr/bin/dkron-processor-syslog
dkron    30896  0.0  1.2  23100 12684 ?        Sl   Oct01   0:00 /usr/bin/dkron-processor-files
dkron    30905  0.0  1.2  23100 12744 ?        Sl   Oct01   0:00 /usr/bin/dkron-processor-log
dkron    30913  0.0  1.2  23120 13100 ?        Sl   Oct01   0:00 /usr/bin/dkron-processor-syslog

```
[...] 
etc

**Comment by yebo29** [comment]
I found a discrepancy in my systemd configs. The node that was _not_ exhibiting the issue is using the systemd unit file provided in the repo, and nothing else. The other two nodes were using a different, more convoluted unit file AND an init file. One of the directives used was to keep processes around even after restart/failure. 

All nodes now using single unit file and cluster has been stable since. Hope this helps others in the future. Closing.
